src/actions/userActions.js

```diff
@@ -135,3 +136,37 @@ export const updateUserAvatarAction = (walletId: string, formData: any) => {
   };
 };
 
+export const labelUserAsLegacyAction = () => {
+  return async (dispatch: Dispatch, getState: GetState, api: Object) => {
+    const {
+      oAuthTokens: { data: oAuthTokens },
+      user: { data: user },
+    } = getState();
+
+    const userWallets = await api.getUserWallets(oAuthTokens.accessToken);
+    if (!userWallets.length) return;
+
+    const keyWallet = userWallets.find(({ type }) => type === ACCOUNT_TYPES.KEY_BASED) || {};
+    const smartWallet = userWallets.find(({ type }) => type === ACCOUNT_TYPES.SMART_WALLET) || {};
```

Comment:
I feel like we could avoid a couple of conditions and make this easier to understand by removing the `|| {}` and just checking if the wallets were found.

For example, it's not very intuitive what would happen if a smart wallet is found but not a key wallet. It looks like the only way to enter on `if (smartWalletCreationTime && diff && diff <= 5) {` is to have both a key wallet and a smart wallet, since the `diff` would be `null` if either is missing, right?

Assuming I understood this correctly, I'd suggest something like this:

```js
    const keyWallet = userWallets.find(({ type }) => type === ACCOUNT_TYPES.KEY_BASED);
    const smartWallet = userWallets.find(({ type }) => type === ACCOUNT_TYPES.SMART_WALLET);
    let isLegacyUser = true;

    if (keyWallet && smartWallet) {
      const { createdAt: keyWalletCreationTime } = keyWallet;
      const { createdAt: smartWalletCreationTime } = smartWallet;

      const diff = Math.floor((new Date(smartWalletCreationTime) - new Date(keyWalletCreationTime)) / 1000 / 60);

      // to those users who gets smart wallet created for them, key based and smart wallets are created one by one
      // in couple of minutes difference
      if (diff <= 5) {
        isLegacyUser = false;
      }
    }

    dispatch({
      type: UPDATE_USER,
      payload: { user: { ...user, isLegacyUser }, state: REGISTERED },
    });
```

Reply:
Well, you can't have a smart wallet if you do not have a key based one :)

Reply:
But yeah, I've revisited this part, and I agree that it could be clearer. I'll push an update in a minute